dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,40 +52,177 @@ public Pattern getParameterPattern() {
+
+        @Override
+        protected ResultSet getObjectMetaData() throws SQLException {
+            return getDBMetaData().getTables(null, name.getSchemaName(), name.getObjectName(), null);
+        }
+
+        @Override
+        public ResultSet getColumns() throws SQLException {
+            return getDBMetaData().getColumns(null, name.getSchemaName(), name.getObjectName(), "%");
+        }
+
+        @Override
+        protected Direction getParamDirection(ResultSet rs) {
+            return Direction.INPUT;
+        }
+    }
+
+    private class DatabaseProcedure extends DatabaseObject {
+
+        private DatabaseProcedure(DatabaseObjectName objName) {
+            super(objName);
+        }
+
+        @Override
+        protected ResultSet getObjectMetaData() throws SQLException {
+            return getDBMetaData().getProcedures(null, name.getSchemaName(), name.getObjectName());
+        }
+
+        @Override
+        public ResultSet getColumns() throws SQLException {
+            return getDBMetaData().getProcedureColumns(null, name.getSchemaName(), name.getObjectName(), "%");
+        }
+
+        @Override
+        protected Direction getParamDirection(ResultSet rs) throws SQLException {
+            switch (rs.getInt("COLUMN_TYPE")) {
+                case DatabaseMetaData.procedureColumnIn:
+                    return Direction.INPUT;
+                case DatabaseMetaData.procedureColumnInOut:
+                    return Direction.INPUT_OUTPUT;
+                case DatabaseMetaData.procedureColumnOut:
+                    return Direction.OUTPUT;
+                default:
+                    throw new SQLException("Unpexpected procedure parameter type for parameter " +
+                                           rs.getString("COLUMN_NAME"));
+            }
+        }
+    }
+
+    private class DatabaseFunction extends DatabaseObject {
+
+        private DatabaseFunction(DatabaseObjectName objName) {
+            super(objName);
+        }
+
+        @Override
+        protected ResultSet getObjectMetaData() throws SQLException {
+            return getDBMetaData().getFunctions(null, name.getSchemaName(), name.getObjectName());
+        }
+
+        @Override
+        public ResultSet getColumns() throws SQLException {
+            return getDBMetaData().getFunctionColumns(null, name.getSchemaName(), name.getObjectName(), "%");
+        }
+
+        @Override
+        protected Direction getParamDirection(ResultSet rs) throws SQLException {
+            int paramType = rs.getInt("COLUMN_TYPE");
+            switch (paramType) {
+                case DatabaseMetaData.functionColumnIn:
+                    return Direction.INPUT;
+                case DatabaseMetaData.functionReturn:
+                    return Direction.RETURN_VALUE;
+                default:
+                    throw new SQLException("Unpexpected function parameter type for parameter " +
+                                  rs.getString("COLUMN_NAME"));
```

Comment:
BTW, initialization logic of static members can be done in static initialiser like:
``` java
static {
    map.put(key, value);
    ...
}
```

Or to do an assignment by call to static method.

But in general static members are not supported in non-static inner classes. If desired in our case - this could rather be done in the parent class.